### ISSUE TITLE
Is it possible require a jsc loader inside a jsc?

### ISSUE THREAD

**Issue by weiluenju** [bytenode/bytenode#172]
I have two js files:

main.js
```
const test = require("./main2.js");
test();
```

main2.js
```
const os = require('os');
module.exports = function(){
  console.log('Hello World!');
  console.log(os.hostname());
}
```

Execute the command

> node main.js

and get the result:

> Hello World!
> DESKTOP-OUNGK9P

Next I compile the js files by the command:

> bytenode -c main.js main2.js

Then replace the main.js content:
```
require('bytenode');
require('./main.jsc');
```

And replace the main2.js content:
```
require('bytenode');
require('./main2.jsc');
```

Then execute the command:

> node main.js

But console show:

> evalmachine.<anonymous>:1
> "​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​"
>                                                                                                      ^
> 
> TypeError: object is not a function
>     at Object.<anonymous> (evalmachine.<anonymous>:1:102)
>     at Object.Module._extensions.<computed> [as .jsc] (D:\bytenode\node_modules\[4mbytenode[24m\lib\index.js:306:26)
> [90m    at Module.load (node:internal/modules/cjs/loader:981:32)[39m
> [90m    at Function.Module._load (node:internal/modules/cjs/loader:822:12)[39m
> [90m    at Module.require (node:internal/modules/cjs/loader:1005:19)[39m
> [90m    at require (node:internal/modules/cjs/helpers:102:18)[39m
>     at Object.<anonymous> (D:\bytenode\main.js:3:5)
> [90m    at Module._compile (node:internal/modules/cjs/loader:1101:14)[39m
> [90m    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)[39m
> [90m    at Module.load (node:internal/modules/cjs/loader:981:32)[39m

Is it possible require a jsc loader inside a jsc?

**Comment by OsamaAbbas** [comment]
You're doing it wrong.

Let's analyze what happens in the end:

main.js (with replaced content) > main.jsc (which expect an exported `test` function) > main2.js (with replaced content, which only requiring main2.jsc, without re-exporting it) > main2.jsc (which exports the `test` function).

Did you see where is the problem?

The minimum change to fix your issue is to change your main2.js to:

```JavaScript
require('bytenode');
module.exports = require('./main2.jsc'); // to export every thing
```

But again, this is very complex and hard to follow. You should find another simpler method of doing things.
